File: app/src/main/java/deltazero/amarok/utils/SecurityUtil.java
package deltazero.amarok.utils;

import deltazero.amarok.AmarokApplication;
import deltazero.amarok.core.Hider;
import deltazero.amarok.core.SettingsSnapshot;

public class SecurityUtil {
  private static boolean locked = true;
  private static boolean disguised = true;

  public static void lockAndDisguise() {
    locked = true;
    disguised = true;
  }

  public static void unlock() {
    locked = false;
  }

  public static void dismissDisguise() {
    disguised = false;
  }

  public static boolean isDisguiseNeeded(AmarokApplication app) {
    SettingsSnapshot settings = app.getSettingsRepo().getSettings().getValue();
    if (settings.getDisableSecurityWhenUnhidden() && Hider.getState() == Hider.State.VISIBLE)
      return false;
    return settings.getDisguise() && disguised;
  }

  public static boolean isUnlockRequired(AmarokApplication app) {
    SettingsSnapshot settings = app.getSettingsRepo().getSettings().getValue();
    if (settings.getDisableSecurityWhenUnhidden() && Hider.getState() == Hider.State.VISIBLE)
      return false;
    return settings.getPassword() != null && locked;
  }
}

File: app/src/main/java/deltazero/amarok/utils/SecurityUtil.kt
package deltazero.amarok.utils

import deltazero.amarok.core.Hider
import deltazero.amarok.core.LockTrigger
import deltazero.amarok.core.SettingsRepository

object SecurityUtil {
  private var locked = true
  private var disguised = true

  @JvmStatic
  fun lockAndDisguise() {
    locked = true
    disguised = true
  }

  /** Re-arm lock and disguise if [trigger] matches the user's configured [LockTrigger]. */
  fun onTrigger(trigger: LockTrigger, settingsRepo: SettingsRepository) {
    if (settingsRepo.settings.value.lockTrigger == trigger) lockAndDisguise()
  }

  @JvmStatic
  fun unlock() {
    locked = false
  }

  @JvmStatic
  fun dismissDisguise() {
    disguised = false
  }

  fun isDisguiseNeeded(settingsRepo: SettingsRepository): Boolean {
    val settings = settingsRepo.settings.value
    if (settings.disableSecurityWhenUnhidden && Hider.getState() == Hider.State.VISIBLE)
      return false
    return settings.disguise && disguised
  }

  fun isUnlockRequired(settingsRepo: SettingsRepository): Boolean {
    val settings = settingsRepo.settings.value
    if (settings.disableSecurityWhenUnhidden && Hider.getState() == Hider.State.VISIBLE)
      return false
    return settings.password != null && locked
  }
}
